### PR TITLE
Test: don't modify defaultConfig on upgrade

### PR DESCRIPTION
**Pull request by stu-elastic**
Otherwise `defaultConfig` on bwc tests is the union of all 
earlier configurations, which may have conflicting settings.

### CODE REVIEW

File: buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java [hunk 1]
```diff
@@ -1095,12 +1096,12 @@ private void createConfiguration() {
...
         }
         // Make sure no duplicate config keys
-        settings.keySet().stream().filter(OVERRIDABLE_SETTINGS::contains).forEach(defaultConfig::remove);
+        settings.keySet().stream().filter(OVERRIDABLE_SETTINGS::contains).forEach(baseConfig::remove);
 
         try {
             Files.write(
                 configFile,
-                Stream.concat(settings.entrySet().stream(), defaultConfig.entrySet().stream())
+                Stream.concat(settings.entrySet().stream(), baseConfig.entrySet().stream())
```
Comment (mark-vieira) [inline buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java:1104]: I'm a bit confused. Is the configuration in `defaultConfig` not needed? `ElasticsearchCluster.commonNodeConfig()` places a bunch of stuff in there. Won't this change now no longer write those config items to the config file or am I missing something?

Comment (stu-elastic) [reply buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java:1104]: On [line 1052](https://github.com/elastic/elasticsearch/pull/55560/files#diff-a704b6e9e08a9ccfcc05e2ae59844495R1052) we initialize `baseConfig` with `defaultConfig`:
`Map<String, String> baseConfig = new HashMap<>(defaultConfig);`

**Review by mark-vieira — APPROVED** [review]
👍

File: buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java (hunk 1 above)
Comment (mark-vieira) [reply buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java:1104]: That's indeed what I was missing.

**Comment by stu-elastic** [comment]
master: 58ec9c3 
7.x (7.8): 41748f0

**Pull request merged**